src/components/MessageList/VirtualizedMessageList.tsx

```diff
@@ -0,0 +1,347 @@
+      <>{EmptyStateIndicator && <EmptyStateIndicator listType='message' />}</>
+    );
+
+    const Header: Components['Header'] = () =>
+      loadingMore ? (
+        <div className='str-chat__virtual-list__loading'>
+          <LoadingIndicator size={20} />
+        </div>
+      ) : (
+        <></>
+      );
+
+    // using 'display: inline-block' traps CSS margins of the item elements
+    // preventing incorrect item measurements.
+    const Item: Components['Item'] = (props) => (
+      <div
+        {...props}
+        style={{
+          display: 'inline-block',
+          width: '100%',
+        }}
+      />
+    );
+
+    const Footer: Components['Footer'] = () =>
+      TypingIndicator ? <TypingIndicator avatarSize={24} /> : <></>;
+
+    return {
+      EmptyPlaceholder,
+      Footer,
+      Header,
+      Item,
+    } as Partial<Components>;
+  }, [EmptyStateIndicator, loadingMore, TypingIndicator]);
+
+  if (!messages) {
+    return null;
+  }
+
+  return (
+    <div className='str-chat__virtual-list'>
+      <Virtuoso
+        atBottomStateChange={(isAtBottom) => {
+          atBottom.current = isAtBottom;
+          if (isAtBottom && newMessagesNotification) {
+            setNewMessagesNotification(false);
+          }
+        }}
+        components={virtuosoComponents}
+        firstItemIndex={PREPEND_OFFSET - numItemsPrepended}
+        followOutput={(isAtBottom) => {
+          if (shouldForceScrollToBottom()) {
+            return isAtBottom ? stickToBottomScrollBehavior : 'auto';
+          }
+          // a message from another user has been received - don't scroll to bottom unless already there
+          return isAtBottom ? stickToBottomScrollBehavior : false;
+        }}
+        initialTopMostItemIndex={
+          messages && messages.length > 0 ? messages.length - 1 : 0
+        }
+        itemContent={(i) => messageRenderer(messages, i)}
+        overscan={overscan}
+        ref={virtuoso}
+        startReached={() => {
+          if (hasMore && loadMore) {
+            loadMore(messageLimit);
+          }
+        }}
+        style={{ overflowX: 'hidden' }}
+        totalCount={messages.length}
+        {...(scrollSeekPlaceHolder
+          ? { scrollSeek: scrollSeekPlaceHolder }
+          : {})}
+      />
+
+      <div className='str-chat__list-notifications'>
+        <MessageNotification
+          onClick={() => {
+            if (virtuoso.current) {
+              virtuoso.current.scrollToIndex(messages.length - 1);
+            }
+            setNewMessagesNotification(false);
+          }}
+          showNotification={newMessagesNotification}
+        >
+          {t('New Messages!')}
+        </MessageNotification>
+      </div>
+    </div>
+  );
+};
+
+/**
+ * The VirtualizedMessageList component renders a list of messages in a virtual list.
+ * It is a consumer of [Channel Context](https://getstream.github.io/stream-chat-react/#channel)
+ * @example ../../docs/VirtualizedMessageList.md
```

Comment:
we should move this markdown file to the same directory as this component and change the path here. remember dealing with this on RN and needed to put it in the same directory for some reason